frontend/src/pages/RoomPage.tsx

```diff
@@ -37,35 +67,135 @@ export const RoomPage = () => {
   }, []);
 
   useEffect(() => {
-    if (leaveRoom && socket) {
-      if (roomId) {
-        socket.emit(ROOM_EVENTS.LEAVE, roomId);
-      }
-      navigate("/dashboard");
+    if (socket && roomId) {
+      socket.emit(ROOM_EVENTS.JOIN, { roomId });
     }
-  }, [leaveRoom, socket, roomId, navigate]);
+  }, [socket, roomId]);
 
   useEffect(() => {
     if (!socket || !roomId) {
       return;
     }
 
-    socket.emit(ROOM_EVENTS.JOIN, roomId);
-
-    socket.on(ROOM_EVENTS.PARTNER_LEAVE, () => {
-      setMessage("The other user has left the room.");
+    socket.on(ROOM_EVENTS.DISCONNECT, (reason: string) => {
+      if (reason === "io server disconnect") {
```

Comment:
This looks like a magic string. Maybe we can create a const for the disconnection reasons.